.github/workflows/sonarcloud.yml

```diff
@@ -0,0 +1,27 @@
+name: SonarCloud
+
+on:
+  workflow_call:
+
+jobs:
+  sonarcloud:
+    name: SonarCloud
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for better relevancy of analysis
+
+      - name: Download artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: lcov.info
+          path: coverage
+
+      - name: SonarCloud Scan
+        # This is SonarSource/sonarcloud-github-action@v2.0.0
+        uses: SonarSource/sonarcloud-github-action@4b4d7634dab97dcee0b75763a54a6dc92a9e6bc1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I believe the secrets need to be explicitly declared at the top here, and passed in from `main.yml`. `workflow_call` action runs don't get any secrets by default.